### PR TITLE
Base URL

### PR DESCRIPTION
**Pull request by andygray**
* [x] Base URL
* [x] Base URL tests
* [x] switched to `edition` to `edition` count

### CODE REVIEW

File: contracts/KnownOriginDigitalAsset.sol [hunk 1]
```diff
@@ -10,7 +11,7 @@ import "./ERC721Token.sol";
 * A curator can mint digital assets and sell them via purchases (crypto via Ether or Fiat)
 */
 contract KnownOriginDigitalAsset is ERC721Token {
-  using SafeMath for uint;
+  using SafeMath for uint256;
```
Comment (jamesmorgan) [inline contracts/KnownOriginDigitalAsset.sol]: Should this not be uint? why change it?

File: contracts/KnownOriginDigitalAsset.sol [hunk 2]
```diff
@@ -71,78 +75,53 @@ contract KnownOriginDigitalAsset is ERC721Token {
...
+    _populateTokenData(_tokenId, _edition, _priceInWei, _auctionStartDate);
   }
 
-  function _populateTokenData(uint _tokenId, bytes16 _edition, uint8 _editionNumber, uint256 _priceInWei, uint32 _auctionStartDate)
-  internal
-  {
+  function _populateTokenData(uint _tokenId, bytes16 _edition, uint256 _priceInWei, uint32 _purchaseFromTime) internal {
     tokenIdToEdition[_tokenId] = _edition;
-    tokenIdToEditionNumber[_tokenId] = _editionNumber;
+    editionToEditionNumber[_edition] = editionToEditionNumber[_edition].add(1);
```
Comment (jamesmorgan) [inline contracts/KnownOriginDigitalAsset.sol]: 👍

File: contracts/KnownOriginDigitalAsset.sol [hunk 3]
```diff
@@ -71,78 +75,53 @@ contract KnownOriginDigitalAsset is ERC721Token {
...
 
-  function burn(uint256 _tokenId)
-  public
-  onlyManagement
-  onlyUnsold(_tokenId)
-  onlyManagementOwnedToken(_tokenId)
-  {
+  function burn(uint256 _tokenId) public onlyManagement onlyUnsold(_tokenId) onlyManagementOwnedToken(_tokenId) {
     // TODO fix me - clean up internal metadata when being burnt
+    // _populateTokenData(_tokenId, 0x0, 0, 0) << will work?
```
Comment (jamesmorgan) [inline contracts/KnownOriginDigitalAsset.sol]: ye maybe, I was thinking maybe we could delete the values altogether (look inside the super call, this is what they do) - not fussed either way though

File: contracts/KnownOriginDigitalAsset.sol [hunk 4]
```diff
@@ -280,85 +234,69 @@ contract KnownOriginDigitalAsset is ERC721Token {
...
+    return Strings.bytes16ToStr(_bytes16, 13, 16);
+  }
+
+  function tokenURI(uint256 _tokenId) public view returns (string) {
+    require(exists(_tokenId));
+    return Strings.strConcat(tokenBaseURI, tokenURIs[_tokenId]);
+  }
+
+  function setTokenBaseURI(string _newBaseURI) external onlyManagement {
+    tokenBaseURI = _newBaseURI;
```
Comment (jamesmorgan) [inline contracts/KnownOriginDigitalAsset.sol]: maybe put a require check on this isnt called with empty string

File: contracts/Strings.sol [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  function strConcat(string _a, string _b, string _c) internal pure returns (string) {
+    return strConcat(_a, _b, _c, "", "");
+  }
+
+  function strConcat(string _a, string _b) internal pure returns (string) {
+    return strConcat(_a, _b, "", "", "");
+  }
+
+  function bytes16ToStr(bytes16 _bytes16, uint8 _start, uint8 _end) internal pure returns (string) {
```
Comment (jamesmorgan) [inline contracts/Strings.sol]: 👍

File: src/store/index.js [hunk 6]
```diff
@@ -173,7 +173,7 @@ const store = new Vuex.Store({
     [actions.GET_ASSETS_PURCHASED_FOR_ACCOUNT] ({commit, dispatch, state}) {
       KnownOriginDigitalAsset.deployed()
         .then((contract) => {
-          return contract.getOwnerTokens(state.account)
+          return contract.tokensOf(state.account)
```
Comment (jamesmorgan) [inline src/store/index.js]: 👍

File: test/truffle/KnownOriginDigitalAsset.test.js [hunk 7]
```diff
@@ -753,76 +754,17 @@ contract('KnownOriginDigitalAsset', function (accounts) {
       });
     });
 
-    describe('mintEdition()', function () {
+    describe.only('purchaseWithEther()', function () {
```
Comment (jamesmorgan) [inline test/truffle/KnownOriginDigitalAsset.test.js]: `.only`

File: contracts/KnownOriginDigitalAsset.sol [hunk 8]
```diff
@@ -71,78 +75,53 @@ contract KnownOriginDigitalAsset is ERC721Token {
...
     commissionAccount = _commissionAccount;
     developerAccount = _developerAccount;
 
+    // FIXME thought? Don't add in constructor and make obvious - ADD AFTER CREATION AS PART OF SETUP?
+    // FIXME as if we change it this will always be the first thing reviewers will see.....?
     // Setup default commission structures
-    tokenIdToCommission["DIG"] = CommissionStructure({curator : 12, developer : 12});
-    tokenIdToCommission["PHY"] = CommissionStructure({curator : 24, developer : 15});
+    editionTypeToCommission["DIG"] = CommissionStructure({curator : 12, developer : 12});
+    editionTypeToCommission["PHY"] = CommissionStructure({curator : 24, developer : 15});
```
Comment (jamesmorgan) [inline contracts/KnownOriginDigitalAsset.sol]: yes makes sense, move to setup js.

File: contracts/KnownOriginDigitalAsset.sol (hunk 1 above)
Comment (andygray) [reply contracts/KnownOriginDigitalAsset.sol]: That is what it is and safe math seems to only work for 256 in my attempts

File: contracts/KnownOriginDigitalAsset.sol (hunk 3 above)
Comment (andygray) [reply contracts/KnownOriginDigitalAsset.sol]: Good point. I think setting to zero is sort of equivalent

**Pull request merged**